proposal/chrisPyr/README.rst

```diff
@@ -0,0 +1,126 @@
+| The kind of problem is actually a N-Body problem. And there is no general solution to N-Body problem, we can only try to simulate it with a numerical method.
+| The simplest idea is to compute all interation force between each body, which is a Brute-force method.
+| By calculating the acceleration and velocity of each time step, we can predict the position of bodies next moment. However, in real life the acceleration and velocity changes even in each time step.
+| Although when time step is small enough, the results should be good enough. But we can try to use "Runge-Kuta Methods"[5] to improve results.
+| In my plan, the final product should be able to show the orbits, periods of planets. And it can be compared with the real world data to prove the correctness and also incorrectness of Newtonian Mechanics.
+| And also do I want to add some intersting feature into it, e.g. users can add/delete planets into/in the system to see how the entire system changes.
+
+
+.. image:: images/Precessing_Kepler_orbit_280frames_e0.6_smaller.gif
+
+Prospective Users
+=================
+
+| This can be used as a supplementary tutorial to high school students and a example to N-body simulation.
+
+System Architecture
+===================
+
+| The default of building the system would create the solar system in a gif or mp4 format. I plan to use Python matplotlib to do the animatin creation part[6].
+| Users can add one or more planets by providing three parameters: intial velocity, position, and mass.
+| To build the solar system I also need information about solar system initial parameters, I plan to get the data from "JPL Horizon(NASA)"[7].
+
+Needed Parameters:
+    - default(build a solar system)
+    - initial acceleration, velocity, position
+    - time step, time period(tunable)
+
+Program Features:
+    - export information of calculated result such as formula of orbit, speed/period of a planet
+    - export a gif or mp4 file to show the drawing result within a time period
+    - add user-defined planets
+    - delete planets
+
+Constraints:
+    - only be able to simulate a period of time, because N-body problem is a chaos system, we don't know the trajectory after our simulation
+    - we cannot perform real integration on computer, so the results may not be perfectly match the real world
+
+System Major Components
+
+| 1. Parameter Parser: Python, check validity of parameters
+| 2. Planet class: C++ class, store information of a planet, including mass, velocity, acceleration, position
+| 3. System class: C++ class, contains all planets object, and prvide operation for computing new position at each time step for each planet
+| 4. ODE class: C++ class, contains several ODE integration methods for system class to call(I will implement Euler Method and Runge-Kutta Method at least)
```

Comment:
By "several", I believe what you intend to do is "one or more".  For the n-body problem, a single numerical method for solving the ODEs should suffice.

It may be interesting to investigate multiple ODE integrators.  It will be a study of the ODE solvers.  A good work, but should take a lot of time from you.